extensions-core/kafka-indexing-service/src/main/java/org/apache/druid/data/input/kafkainput/KafkaInputReader.java

```diff
@@ -173,14 +174,24 @@ private CloseableIterator<InputRow> buildBlendedRows(
         r -> {
           final MapBasedInputRow valueRow;
           try {
-            // Return type for the value parser should be of type MapBasedInputRow
-            // Parsers returning other types are not compatible currently.
-            valueRow = (MapBasedInputRow) r;
+            if (r instanceof ListBasedInputRow) {
```

Comment:
I think the performance implication of this if should be okay. 
Anyway can we add a UT for this ?